src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -105,7 +105,10 @@ class AdvertiserMinMdns : public ServiceAdvertiser,
     AdvertiserMinMdns() : mResponseSender(&GlobalMinimalMdnsServer::Server())
     {
         GlobalMinimalMdnsServer::Instance().SetQueryDelegate(this);
-        mResponseSender.AddQueryResponder(mQueryResponderAllocatorOperational.GetQueryResponder());
+        for (size_t i = 0; i < kMaxOperationalNetworks; ++i)
```

Comment:
Will resolution of https://github.com/project-chip/connectedhomeip/issues/8000 allow us to get away from this hard-coded bound?

Reply:
Sort of - plan right now is to template it, so the bound will be settable at compile time, but not adjustable on the fly. Did you mean you want to support an arbitrary number of operational networks? Or just have the max number be settable?

Reply:
I am proposing this should be arbitrary and determined at runtime.

The notion of Fabric Index, and therefore a bound to the number of supported fabrics, is really only relevant for nodes that choose to expose the `Operational Credentials` cluster.  This likely does not make sense for commissioners and other nodes that must be bootstrapped to fabrics by out-of-band means.  And note that all fabrics require at least one 'originating' node that has been bootstrapped to the fabric by out-of-band means.

Such nodes can theoretically be joined to an arbitrary number of fabrics, only bound by their available memory and the scalability of mDNS.  A compile-time bound doesn't really support these types of nodes very well.

In fairness, this is really a larger problem in the `AdminPairingTable` object and the sdk's reliance on this.  Code paths that rely on this object are moving us toward a design that really only makes sense for nodes that expose the `Operational Credentials` cluster.

Reply:
Yeah, I think this is actually coming into a larger discussion around run-time memory allocation vs. static allocation and the differing requirements of the smaller (MPU / no mmu / limited memory) devices and the more capable devices with all the bells and whistles. As much as I prefer not to have a proliferation of compile-flags for testing purposes, we may want to consider allowing device manufacturers to select between a pre-allocation (and its size) and a dynamic allocation scheme. #8000 doesn't cover that though because that's larger than just mdns.

Reply:
OK, fair enough.

In any case, this PR obviously moves us at least toward properly supporting the `kMaxOperationalNetworks` the sdk accommodates today.  So this makes sense.

Thank you.

Reply:
K, going to resolve this comment for now, and continue discussion on #8000.